source/fundamentals/logging.txt

```diff
@@ -0,0 +1,434 @@
+the ``LogSink`` for your ``Client`` by calling the ``SetSink()`` method on
+your ``LoggerOptions`` instance.
+
+Example
+~~~~~~~
+
+This example demonstrates how to define and implement a custom logger.
+
+.. procedure::
+
+   .. step:: Define the ``CustomLogger`` struct
+
+      .. literalinclude:: /includes/fundamentals/code-snippets/logging.go
+         :language: go
+         :dedent:
+         :start-after: start-customlogger-struct
+         :end-before: end-customlogger-struct
+
+   .. step:: Implement the ``Info()`` and ``Error()`` methods with custom log message formatting
+
+      .. literalinclude:: /includes/fundamentals/code-snippets/logging.go
+         :language: go
+         :dedent:
+         :start-after: start-customlogger-funcs
+         :end-before: end-customlogger-funcs
+
+   .. step:: Assign a ``Writer`` to your logger and set it as the ``Sink`` for your ``Client``
+
+      In this example, the logger logs against commands and connection
+      events at the ``DEBUG`` level:
+
+      .. literalinclude:: /includes/fundamentals/code-snippets/logging.go
+         :language: go
+         :dedent:
+         :emphasize-lines: 1-2, 4-8, 13
+         :start-after: start-set-customlogger
+         :end-before: end-set-customlogger
+
+   .. step:: Perform an operation
+
+      The following code performs an insert operation, which generates log
+      messages:
+
+      .. io-code-block::
+         :copyable: true
+
+         .. input:: /includes/fundamentals/code-snippets/logging.go
+            :language: go
+            :dedent:
+            :start-after: start-insert
+            :end-before: end-insert
+
+         .. output::
+            :language: console
+            :visible: false
+
+            level: 1 DEBUG, message: Connection pool created
+            level: 1 DEBUG, message: Connection pool ready
+            level: 1 DEBUG, message: Connection pool created
+            level: 1 DEBUG, message: Connection pool ready
+            level: 1 DEBUG, message: Connection pool created
+            level: 1 DEBUG, message: Connection pool ready
+            level: 1 DEBUG, message: Connection checkout started
+            level: 1 DEBUG, message: Connection created
+            level: 1 DEBUG, message: Connection ready
+            level: 1 DEBUG, message: Connection checked out
+            level: 1 DEBUG, message: Command started
+            level: 1 DEBUG, message: Command succeeded
+            level: 1 DEBUG, message: Connection checked in
+
+.. _golang-logging-thirdparty:
+
+Integrate Third-Party Loggers
+-----------------------------
+
+There are many third-party logging packages available in Go. To use
+a third-party logger in your application, create a logger and assign it
+as the sink in your ``LoggerOptions`` instance.
+
+Example
+~~~~~~~
+
+This example demonstrates how to define and integrate a :github:`logrus <sirupsen/logrus>`
+logger in your application.
```

Comment:
We probably don't want to interrupt this example by sending them to GitHub

S: Remove link here and modify admonition below

```suggestion
This example demonstrates how  to integrate ``logrus``,
a third-party logging package, into your application.
```